### ISSUE TITLE
(🐞) Ruff doesn't handle spaces in file names when run through python

### ISSUE THREAD

**Issue by KotlinIsland** [astral-sh/ruff#5077]
```
> ruff "a b.py"
> python -m ruff "a b.py"
error: Failed to lint a: The system cannot find the file specified. (os error 2)
error: Failed to lint b.py: The system cannot find the file specified. (os error 2)
a:1:1: E902 The system cannot find the file specified. (os error 2)
b.py:1:1: E902 The system cannot find the file specified. (os error 2)
Found 2 errors.
```

Windows 10
powershell 7.4 / 5.1 and CMD (and I think invoking it programmatically as well)
ruff 0.0.272
python 3.11.2

**Comment by zanieb** [comment]
@KotlinIsland / @konstin isn't this just because double quotes are used? `python -m ruff 'a b.py'` works as intended, as does `python -m ruff a\ b.py`.

**Comment by konstin** [comment]
I can't reproduce this with PowerShell 7.3.4 on ubuntu 22.04 (`python -m ruff "a b.py"`) works, if it doesn't resolve itself until then i'll boot into windows and check tomorrow

**Comment by KotlinIsland** [comment]
using an escape character for the space showed the same result

Works
```
ruff a` b.py
```

Doesn't work
```
python -m ruff a` b.py
```

**Comment by konstin** [comment]
I could reproduce this when using windows instead of linux and fixed this in #5115
